File: engine/engine-model/src/main/scala/tools/packet/guild/GuildMemberLeft.scala
package tools.packet.guild

import net.opcodes.SendOpcode
import tools.packet.PacketInput

class GuildMemberLeft(private var _guildId: Int, private var _characterId: Int, private var _characterName: String,
                      private var _expelled: Boolean) extends PacketInput {
  def guildId: Int = _guildId

  def characterId: Int = _characterId

  def characterName: String = _characterName

  def expelled: Boolean = _expelled

  override def opcode(): SendOpcode = SendOpcode.GUILD_OPERATION
}

File: engine/engine-model/src/main/java/tools/packet/guild/GuildMemberLeft.java
package tools.packet.guild;

import net.opcodes.SendOpcode;
import tools.packet.PacketInput;

public record GuildMemberLeft(Integer guildId, Integer characterId, String characterName,
                              Boolean expelled) implements PacketInput {
   @Override
   public SendOpcode opcode() {
      return SendOpcode.GUILD_OPERATION;
   }
}